src/managers/KeybindManager.cpp

```diff
@@ -454,11 +454,33 @@ void CKeybindManager::changeworkspace(std::string args) {
     int workspaceToChangeTo = 0;
     std::string workspaceName = "";
 
+    // Flag needed so that the previous workspace is not recorded when switching
+    // to a previous workspace.
+    bool isSwitchingToPrevious = false;
+
     if (args.find("[internal]") == 0) {
         workspaceToChangeTo = std::stoi(args.substr(10));
         const auto PWORKSPACE = g_pCompositor->getWorkspaceByID(workspaceToChangeTo);
         if (PWORKSPACE)
             workspaceName = PWORKSPACE->m_szName;
+    } else if (args.find("previous") == 0) {
+        const auto P_CURRENT_WORKSPACE = g_pCompositor->getWorkspaceByID(
```

Comment:
code style, no snake case

Reply:
Will do, although I'd point out that the all-caps variables can get a bit hard to read.